src/syscheckd/src/create_db.c

```diff
@@ -265,34 +271,27 @@ time_t fim_scan() {
 
     w_mutex_unlock(&syscheck.fim_scan_mutex);
 
+    if (syscheck.db_entry_limit_enabled) {
+        nodes_count = fim_db_get_count_file_entry();
+    }
 
     fim_db_transaction_deleted_rows(db_transaction_handle, transaction_callback, &txn_ctx);
     db_transaction_handle = NULL;
 
-#ifdef WIN32
-    fim_registry_scan();
-#endif
-    if (syscheck.db_entry_limit_enabled) {
-        /* DEPRECATED CODE
-        nodes_count = fim_db_get_count_entries(syscheck.database);
-        */
-    }
-
     if (syscheck.db_entry_limit_enabled && (nodes_count >= syscheck.db_entry_file_limit)) {
         w_mutex_lock(&syscheck.fim_scan_mutex);
 
         db_transaction_handle = fim_db_transaction_start(FIMDB_FILE_TXN_TABLE, transaction_callback, &txn_ctx);
 
         w_rwlock_rdlock(&syscheck.directories_lock);
         OSList_foreach(node_it, syscheck.directories) {
+            if (txn_ctx.db_full == true) {
+                break;
+            }
+
             dir_it = node_it->data;
             char *path;
             event_data_t evt_data = { .mode = FIM_SCHEDULED, .report_event = true, .w_evt = NULL };
-            /* DEPRECATED CODE
-            if (fim_db_is_full(syscheck.database)) {
```

Comment:
If we definitely remove the function `fim_db_is_full`. Could we remove it from the unit tests as well?